### PR TITLE
First version of useEffect working

### PR DESCRIPTION
**Pull request by jduthon**
This PR adds `useEffect`, it was suprisingly simple as most of the architecture for the hooks was already in place from the useState one 🤩 

We can walk through it maybe on our next SU and I believe you will walk me through your diff changes :)

I went for it to be all blocking calls, I am not sure whether that's what React does or not, chances are they try to be a bit smarter with how they apply the effects, but I think for what we want for the workshop, that might be enough. :)

### CODE REVIEW

**Comment by jduthon** [comment]
EDIT: this is missing an important part: running the clean-up effect returned by the effect provided by consumers 🙈 

Will work on getting that today.

File: packages/my-own-react/index.js [hunk 1]
```diff
@@ -124,20 +125,64 @@ const makeMakeUseState = (onUpdate, hooksMap) => (VDOMPointer, isFirstRender) =>
...
+    }
+    return (effectCallback, dependencies) => {
+      const effectIndex = effectIndexRef.current;
+      const previousEffect = hooksMapPointer.effect[effectIndex] || {};
+      const { cleanUp = (() => {})} = previousEffect;
+      effectIndexRef.current += 1;
+      if (areDependenciesEqual(previousEffect.dependencies, dependencies)) {
+        return;
+      }
+      cleanUp();
```
Comment (jduthon) [inline packages/my-own-react/index.js:157]: This will not work for unmounting at the moment.

Comment (jduthon) [reply packages/my-own-react/index.js:187]: This is done now 🎉

**Comment by jduthon** [comment]
@Simba14 @omergulen took care of merge conflicts here as well as the open comment I had about cleaning up hooks for unmounted components, so this is also ready for review :)

**Review by omergulen — APPROVED** [review]
LGTM! 🥳

**Pull request merged**